src/main/java/com/avioconsulting/mule/logger/internal/config/CustomLoggerConfiguration.java

```diff
@@ -130,6 +154,18 @@ public boolean isEnableV1Compatibility() {
     return enableV1Compatibility;
   }
 
+  public Compressor getCompressor() {
+    return compressor;
+  }
+
+  public EncryptionAlgorithm getEncryptionAlgorithm() {
+    return encryptionAlgorithm;
+  }
+
+  public String getEncryptionPassword() {
+    return encryptionPassword;
+  }
```

Comment:
We should validate these configuration values to ensure either both or none are set. When one is provided, the other must also be provided since we are only supporting JCE with Password-based. `org.mule.runtime.api.lifecycle.Initialisable` can be implemented to validate configuration in `initialise` method.

Reply:
Implementation added